ruff_lsp/server.py

```diff
@@ -550,13 +835,16 @@ async def code_action(params: CodeActionParams) -> list[CodeAction] | None:
     # Add "Disable for this line" for every diagnostic.
     if settings.get("codeAction", {}).get("disableRuleComment", {}).get("enable", True):
         if not params.context.only or CodeActionKind.QuickFix in params.context.only:
+            document = Document.from_text_document(
+                LSP_SERVER.workspace.get_text_document(params.text_document.uri)
+            )
             lines: list[str] | None = None
             for diagnostic in params.context.diagnostics:
                 if diagnostic.source == "Ruff":
                     noqa_row = cast(DiagnosticData, diagnostic.data).get("noqa_row")
                     if noqa_row is not None:
                         if lines is None:
-                            lines = document.lines
+                            lines = document.source.splitlines(keepends=True)
```

Comment:
Is this _definitely_ equivalent?

Reply:
Are you asking w.r.t. the original implementation? If so, yes.

If you're asking w.r.t. the `document` variable, then yes as well because we're using the text document version for this case which is either a Python file or a Notebook cell.

Reply:
I've reverted this to use the `pygls` implementation